tests/sharding/test.sh

```diff
@@ -0,0 +1,300 @@
+}
+
+TEST_validate_shard1_active_for_first_validity_window() {
+  local id=1.1
+  local artifact_path=$WORKDIR/data.$id.txt
+  make_artifact $artifact_path
+  go run client/sign/sign.go -bundle-out $WORKDIR/bundle.$id.json -key-out $WORKDIR/entry-key.$id.pem -trusted-root $WORKDIR/targets/trusted_root.json -signing-config $WORKDIR/targets/signing_config.json $artifact_path
+  assert_log_index $WORKDIR/bundle.$id.json 1
+  assert_shard $WORKDIR/bundle.$id.json shard1.rekor.local
+  go run client/verify/verify.go -bundle $WORKDIR/bundle.$id.json -key $WORKDIR/entry-key.$id.pem -trusted-root $WORKDIR/targets/trusted_root.json $artifact_path
+}
+
+TEST_validate_shard1_entry_for_second_validity_window() {
+  local id=1.1
+  local artifact_path=$WORKDIR/data.$id.txt
+  go run client/verify/verify.go -bundle $WORKDIR/bundle.$id.json -key $WORKDIR/entry-key.$id.pem -trusted-root $WORKDIR/targets/trusted_root.json $artifact_path
+}
+
+TEST_validate_shard2_active_for_second_validity_window() {
+  local id=2.1
+  local artifact_path=$WORKDIR/data.$id.txt
+  make_artifact $artifact_path
+  go run client/sign/sign.go -bundle-out $WORKDIR/bundle.$id.json -key-out $WORKDIR/entry-key.$id.pem -trusted-root $WORKDIR/targets/trusted_root.json -signing-config $WORKDIR/targets/signing_config.json $artifact_path
+  assert_log_index $WORKDIR/bundle.$id.json 1
+  assert_shard $WORKDIR/bundle.$id.json shard2.rekor.local
+  go run client/verify/verify.go -bundle $WORKDIR/bundle.$id.json -key $WORKDIR/entry-key.$id.pem -trusted-root $WORKDIR/targets/trusted_root.json $artifact_path
+}
+
+run() {
+  # Set up TSA.
+  message "SETUP: Starting TSA"
+  start_tsa
+  local tsakey=$(get_tsa_cert)
+
+  # Bring up the first shard. Set up trusted_root.json and signing_config.json
+  # to point to it. Signing and verifying should work normally.
+  message "SETUP: starting first shard"
+  local shard1_key=$(new_key 1)
+  start_shard1 $tsakey $shard1_key
+
+  # Test that the one service works.
+  SETUP_validate_shard1_initialized
+
+  # Bring up the second shard. Add the second shard to trusted_root.json and
+  # signing_config.json to be chosen for signing in the future but be valid for
+  # verification now. Signing should upload entries to the first shard, but
+  # verification should work for both.
+  message "SETUP: starting second shard"
+  local shard2_key=$(new_key 2)
+  start_shard2 $tsakey $shard1_key $shard2_key
+  SETUP_validate_shard2_initialized
+
+  # Test that the second service works, using its URL explicitly, and check that
+  # verifying works with the new TUF keys.
+  # Test that the original service is still selected by signing config and still works.
+  message "TEST: shard 1 works during first validity window"
+  TEST_validate_shard1_active_for_first_validity_window
+
+  message "WAIT: sleeping for 65 seconds to move to new validity window"
+  sleep 65
```

Comment:
this is probably a dumb question, can we fake passage of time? like change system clock? or something? That might be too dangerous/destructive.

Reply:
Turns out yes, there's a thing called `faketime` that I switched this to use.